src/components/elements/FlexibleGatewayCard.tsx

```diff
@@ -0,0 +1,121 @@
+import { useRouter } from "next/router";
+import { Card, Col, Row, Typography } from "antd";
+import NodeCard from "./FlexibleNodeCard";
+import {
+  getFormattedLastSeen,
+  getFormattedLastSeenDate,
+  getFormattedVoltageData,
+} from "../presentation/uiHelpers";
+import { GATEWAY_MESSAGE, ERROR_MESSAGE } from "../../constants/ui";
+import styles from "../../styles/Home.module.scss";
+import cardStyles from "../../styles/Card.module.scss";
+import { Gateway, SensorTypeCurrentReading, GatewaySensorTypeNames, NodeSensorTypeNames } from "../../services/AppModel";
+
+interface GatewayProps {
+  gateway: Gateway;
+  index: number;
+}
+
+function findCurrentReadingWithName(gateway: Gateway, name: SensorTypeCurrentReading["sensorType"]["name"]) : SensorTypeCurrentReading | undefined{
+  return gateway.currentReadings?.find((sensorTypeReading) => name===sensorTypeReading.sensorType.name);
+}
```

Comment:
This should probably be moved either into the body of this component or into a separate helper file and then called here, based on if other components will need to access it too

Reply:
Let's iterate on it after this PR is merged.